packages/angular/src/builders/webpack-server/webpack-server.impl.ts

```diff
@@ -0,0 +1,137 @@
+import { BuilderContext, createBuilder } from '@angular-devkit/architect';
+import { JsonObject } from '@angular-devkit/core';
+import { joinPathFragments, readCachedProjectGraph } from '@nrwl/devkit';
+import {
+  calculateProjectDependencies,
+  checkDependentProjectsHaveBeenBuilt,
+  createTmpTsConfig,
+  DependentBuildableProjectNode,
+} from '@nrwl/workspace/src/utilities/buildable-libs-utils';
+import { existsSync } from 'fs';
+import { join } from 'path';
+import { Observable, of } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+import { merge } from 'webpack-merge';
+import { resolveCustomWebpackConfig } from '../utilities/webpack';
+import { WebpackServerSchema } from './schema';
+import {
+  executeServerBuilder,
+  ServerBuilderOptions,
+  ServerBuilderOutput,
+} from '@angular-devkit/build-angular';
+
+type Schema = ServerBuilderOptions & WebpackServerSchema;
+
+function buildServerApp(
+  options: Schema,
+  context: BuilderContext
+): Observable<ServerBuilderOutput> {
+  const { buildLibsFromSource, customWebpackConfig, ...delegateOptions } =
+    options;
+  // If there is a path to custom webpack config
+  // Invoke our own support for custom webpack config
+  if (customWebpackConfig && customWebpackConfig.path) {
+    const pathToWebpackConfig = joinPathFragments(
+      context.workspaceRoot,
+      customWebpackConfig.path
+    );
+
+    if (existsSync(pathToWebpackConfig)) {
+      return buildServerAppWithCustomWebpackConfiguration(
+        delegateOptions,
+        context,
+        pathToWebpackConfig
+      );
+    } else {
+      throw new Error(
+        `Custom Webpack Config File Not Found!\nTo use a custom webpack config, please ensure the path to the custom webpack file is correct: \n${pathToWebpackConfig}`
+      );
+    }
+  }
+
+  return executeServerBuilder(delegateOptions, context);
+}
+
+function buildServerAppWithCustomWebpackConfiguration(
+  options: Schema,
+  context: BuilderContext,
+  pathToWebpackConfig: string
+) {
+  return executeServerBuilder(options, context as any, {
+    webpackConfiguration: async (baseWebpackConfig) => {
+      const customWebpackConfiguration = resolveCustomWebpackConfig(
+        pathToWebpackConfig,
+        options.tsConfig
+      );
+      // The extra Webpack configuration file can also export a Promise, for instance:
+      // `module.exports = new Promise(...)`. If it exports a single object, but not a Promise,
+      // then await will just resolve that object.
+      const config = await customWebpackConfiguration;
+
+      // The extra Webpack configuration file can export a synchronous or asynchronous function,
+      // for instance: `module.exports = async config => { ... }`.
+      if (typeof config === 'function') {
+        return config(baseWebpackConfig, options, context.target);
+      } else {
+        return merge(baseWebpackConfig, config);
+      }
```

Comment:
We do this in a couple of places, we could later consider extracting it into a helper. Something for a different cleanup PR if we think it is worthy.

Reply:
Yes, can be a follow up PR for a utility to manage this :)